sovryn-token-bridge/federator/src/lib/GasServices.js

```diff
@@ -0,0 +1,97 @@
+        this.logger = logger;
+        this.config = config;
+        
+        // this.gasPriceEstimator = new GasPriceEstimator({
+        //     web3: this.client,
+        //     logger: this.logger,
+        //     etherscanApiKey: config.etherscanApiKey,
+        // });
+
+
+        this.gasPriceFetcher = new GasPriceFetcher(
+            log4js.getLogger('ETH-MAINNET-GasPriceFetcher'),
+            config.etherscanApiKey
+        );
+        
+        this.gasPriceAvg = new GasPriceAvg(
+            log4js.getLogger('ETH-MAINNET-GasPriceFetAvg'),
+        );
+    }
+    
+    async startGasServices() {
+        let etherScanGasPriceService = new Scheduler(gasApiPollingInterval, this.logger, { run: () => this.runGasPriceService() });
+        let gasPriceAvgService = new Scheduler(avgGasPollingInterval, this.logger, { run: () => this.runGasPriceAvg() });
+        
+        if(avgGasPollingInterval > 0) {
+            avgGasCount = avgGasPeriodInterval/avgGasPollingInterval;
+        }
+        else {
+            this.logger.error('config.avgGasPollingInterval cannot be 0', err);
+            process.exit();
+        };
+
+        etherScanGasPriceService.start().catch((err) => {
+            this.logger.error('Unhandled Error on etherScanGasPriceService start()', err);
+        });
+
+        await utils.sleep(2 * gasApiPollingInterval, { logger: this.logger });
+
+        gasPriceAvgService.start().catch((err) => {
+            this.logger.error('Unhandled Error on gasPriceAvg start()', err);
+        });  
+
+        this.logger.info(`Process is waiting to calculate average gas of ${avgGasPeriodInterval} ms.`);
+        //log4js.getLogger('GAS-SERVICE-WAKEUP');
+        //console.log(`Process is waiting to calculate average gas of ${avgGasPeriodInterval} ms.`);
+        this.logger.debug('Process is waiting to calculate average gas of ' ,avgGasPeriodInterval);
+        await utils.sleep(avgGasPeriodInterval, { logger: this.logger });
+    }
+
+    async runGasPriceService() {
+        
+        try {
+            globals.currentEthGasBasePrice = await this.gasPriceFetcher.getEtherscanBaseGasPrice();
+            this.logger.debug('currentEthGasBasePrice: ' ,globals.currentEthGasBasePrice);
+            console.log("currentEthGasBasePrice: " + globals.currentEthGasBasePrice);
+        } catch(err) {
+            this.logger.error('Unhandled Error on runGasPriceService()', err);
+            process.exit();
+        }
+    }
+    
+    async runGasPriceAvg() {
+        
+        try {
+            globals.currentEthGasPriceAvg = await this.gasPriceAvg.calcAvg(avgGasCount, globals.currentEthGasBasePrice);
+            this.logger.debug('currentEthGasPriceAvg: ' ,globals.currentEthGasPriceAvg);
+            console.log("currentEthGasPriceAvg: " + globals.currentEthGasPriceAvg);
+        } catch(err) {
+            logger.error('Unhandled Error on runGasWatcher()', err);
+            process.exit();
```

Comment:
As above, I think we should not exit here after all, since it will mean that the federator will crash on a http error instead of retrying.

Reply:
Fixed